lib/rgeo/shapefile/reader.rb

```diff
@@ -342,7 +343,10 @@ def next
       # and yield the Reader::Record for each one.
 
       def each
+        return to_enum(:each) unless block_given?
         return unless @opened
+        # Each needs to be idempotent, therefore we reset all the internal indexes to their original value
+        rewind
         yield _read_next_record while @cur_record_index < @num_records
```

Comment:
Since a user may now call every enumerable method, and still use the original `next` based api (which might be deprecated then ? @keithdoggett). I think we should ensure that `each` will keep users current index :

```suggestion
        # Each needs to be idempotent, therefore we reset all the internal indexes to their original value
        current_record_index = @cur_record_index
        rewind
        yield _read_next_record while @cur_record_index < @num_records
      ensure
        current_record_index && @cur_record_index = current_record_index
```

Reply:
Shoult not  it be
`current_record_index && seek_index(current_record_index)` 
?

Reply:
Definitely! Moreover, I don't know what is the behavior when `seek_index` returns `false` but it may mean that it is not ok to continue then